src/boot/pprint.ml

```diff
@@ -326,20 +320,25 @@ and print_tm' fmt t = match t with
     let inner = List.map print (Mseq.to_list tms) in
     fprintf fmt "[@[<hov 0>%a@]]" concat (Comma,inner)
 
-  | TmTuple(_,tms) ->
-    let print t = (fun fmt -> fprintf fmt "%a" print_tm (App,t)) in
-    let inner = List.map print tms in
-    fprintf fmt "(@[<hov 0>%a@])" concat (Comma,inner)
-
   | TmRecord(_,r) ->
-    let contents = Record.fold (fun l v ack -> (l, v)::ack) r [] in
-    print_record fmt contents
-
-  | TmProj(_,t,l) ->
-    let l = match l with
-      | LabIdx i -> string_of_int i
-      | LabStr s -> string_of_ustring s in
-    fprintf fmt "%a.%s" print_tm (Atom, t) l
+    if r = Record.empty then fprintf fmt "()"
+    else
+      let parsed_record_to_tuple =
+        List.fold_left (fun (a,k) (i,tm) ->
+          match a with
+          | Some _ when (try (int_of_ustring i) != k with _ -> true) -> (None,0)
+          | Some lst -> (Some(tm::lst),k+1)
+          | None -> (None,0)) (Some [], 0) (Record.bindings r)
```

Comment:
These parentheses feel a bit confusing

Reply:
Note sure which parentheses you mean, but the last two are arguments to the `fold_left` function (you cannot see the fold in the code snippet above). It does change the meaning of the code, but I separated the fold function into a separate named function. This might be easier to read...